src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -266,14 +269,12 @@ def _normalize(
         # If user defined their udf as @udf(lambda: x, ...) then extension_fn.handler is <lambda>.
         extension_fn.handler = f"{py_file.stem}.{extension_fn.handler}"
 
-        snowpark_lib_found = False
-        snowpark_lib_regex = re.compile(SNOWPARK_LIB_REGEX)
-        for pkg in extension_fn.packages:
-            if snowpark_lib_regex.fullmatch(ensure_string_literal(pkg.strip())):
-                snowpark_lib_found = True
-                break
-        if not snowpark_lib_found:
-            extension_fn.packages.append(SNOWPARK_LIB_NAME)
+        extension_fn.packages = [
+            self._normalize_package_name(pkg) for pkg in extension_fn.packages
+        ]
```

Comment:
this is replacing all package names with the normalized version, which the current code doesn't do. Is that intentional?

Reply:
Yes, this is done so I can reliably look for the snowpark lib package in the next step. We always did that as a last step (which I didn't change in this PR because we have tests that verify that this happens as a safety net).